### PR TITLE
add debugging on scale failure

### PR DESCRIPTION
**Pull request by jim-minter**
```release-note
NONE
```

### CODE REVIEW

**Comment by asalkeld** [comment]
flake #552 
/retest

**Comment by charlesakalugwu** [comment]
flake #552 
/retest

File: pkg/cluster/scaler/scaler.go [hunk 1]
```diff
@@ -116,6 +120,24 @@ func (ws *workerScaler) scaleUp(ctx context.Context, count int64) *api.PluginErr
...
+		spew.Dump(w, err)
+		// dump the VMSS state
+		ss, err2 := ws.ssc.Get(ctx, ws.resourceGroup, *ws.ss.Name)
+		spew.Dump(w, ss, err2)
+		// try the scale again
+		err2 = ws.ssc.Update(ctx, ws.resourceGroup, *ws.ss.Name, azcompute.VirtualMachineScaleSetUpdate{
+			Sku: &azcompute.Sku{
+				Capacity: to.Int64Ptr(count),
+			},
+		})
```
Comment (ehashman) [inline pkg/cluster/scaler/scaler.go]: Is it possible for this to succeed so then we shouldn't fail below?

File: pkg/util/azureclient/compute/virtualmachinescalesets.go [hunk 2]
```diff
@@ -30,7 +30,6 @@ var _ VirtualMachineScaleSetsClient = &virtualMachineScaleSetsClient{}
 func NewVirtualMachineScaleSetsClient(ctx context.Context, log *logrus.Entry, subscriptionID string, authorizer autorest.Authorizer) VirtualMachineScaleSetsClient {
 	client := compute.NewVirtualMachineScaleSetsClient(subscriptionID)
 	azureclient.SetupClient(ctx, log, "compute.VirtualMachineScaleSetsClient", &client.Client, authorizer)
-	client.PollingDuration = 30 * time.Minute
```
Comment (ehashman) [inline pkg/util/azureclient/compute/virtualmachinescalesets.go]: Why is this duration being removed? Does it have a sensible default we're overriding?

File: pkg/cluster/scaler/scaler.go (hunk 1 above)
Comment (jim-minter) [reply pkg/cluster/scaler/scaler.go]: yes, absolutely possible that it succeeds - for now I'd like to see the evidence of that, then will code up the retrier

File: pkg/util/azureclient/compute/virtualmachinescalesets.go (hunk 2 above)
Comment (jim-minter) [reply pkg/util/azureclient/compute/virtualmachinescalesets.go]: 15 minutes.  Previously I tried upping it to see if that would reduce our issues, but I don't think it did.

**Pull request merged**